### PR TITLE
[BugFix] Fix AWS signature mismatch when querying partitioned Hive ta…

### PR DESCRIPTION
**Pull request by gddezero**
…ble on GCS (#18912)

## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes (https://github.com/StarRocks/starrocks/issues/18912)

## Problem Summary(Required) ：

Described in (https://github.com/StarRocks/starrocks/issues/18912)
**Note: Need to build thirdparty libraries and also update thirdparty libs in docker images used to build fe/be.**

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by imay** [comment]
@gddezero thank you for contribution, could you sign off by “git commit -s”

File: be/CMakeLists.txt [hunk 1]
```diff
@@ -211,6 +211,9 @@ message(STATUS "Using OpenSSL Root Dir: ${OPENSSL_ROOT_DIR}")
 add_library(crypto STATIC IMPORTED)
 set_target_properties(crypto PROPERTIES IMPORTED_LOCATION ${THIRDPARTY_DIR}/lib/libcrypto.a)
 
+add_library(AWS::crypto STATIC IMPORTED)
```
Comment (imay) [inline be/CMakeLists.txt:214]: Why did you add this library? I don't see how it is used?

**Comment by imay** [comment]
I suggest you split this PR into two. One is for third-party, one is for the other

File: be/CMakeLists.txt (hunk 1 above)
Comment (gddezero) [reply be/CMakeLists.txt:214]: After aws-sdk-cpp is upgraded from 1.9.179 to 1.0.272, it will fail with the following error when building thirdparty libraries:
```
CMake Error at src/service/CMakeLists.txt:49 (add_executable):
  Target "starrocks_be" links to target "AWS::crypto" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?
```
My solution is to add library: AWS::crypto. But if there is other better approches, I will be very happy to learn and adopt.

Comment (imay) [reply be/CMakeLists.txt:214]: Did you try to add crypto in find package of awssdk?

Comment (gddezero) [reply be/CMakeLists.txt:214]: Let me try it. Thanks!

**Comment by gddezero** [comment]
> I suggest you split this PR into two. One is for third-party, one is for the other
Sure, will close this PR and create 2 new PRs for third-party and be seperately.

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5
